### PR TITLE
PAPP-24382: Initial release

### PR DESCRIPTION
**Pull request by ebhoir-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: msadgraph.json [hunk 1]
```diff
@@ -0,0 +1,2969 @@
...
+    "appid": "f2a239df-acb2-47d6-861c-726a435cfe76",
+    "name": "MS Graph for Active Directory",
+    "description": "Connects to Microsoft AD using MS Graph REST API services",
+    "type": "identity management",
+    "product_vendor": "Microsoft",
+    "logo": "logo_msadgraph_dark.svg",
+    "logo_dark": "logo_msadgraph_dark.svg",
+    "product_name": "MS Graph for Active Directory",
+    "product_version_regex": ".*",
+    "publisher": "Splunk Community",
```
Comment (dgopani-crest) [inline msadgraph.json:11]: "publisher": "Splunk"

File: msadgraph.json [hunk 2]
```diff
@@ -0,0 +1,2969 @@
...
+    "contributors": [
+        {
+            "name": "Ekta Bhoir"
+        }
+    ],
+    "license": "Copyright (c) 2019-2022 Splunk Inc.",
```
Comment (dgopani-crest) [inline msadgraph.json:17]: As this is a new release, it should be Copyright (c) 2022 Splunk Inc.

File: msadgraph.json [hunk 3]
```diff
@@ -0,0 +1,2969 @@
...
+    "name": "MS Graph for Active Directory",
+    "description": "Connects to Microsoft AD using MS Graph REST API services",
+    "type": "identity management",
+    "product_vendor": "Microsoft",
+    "logo": "logo_msadgraph_dark.svg",
+    "logo_dark": "logo_msadgraph_dark.svg",
+    "product_name": "MS Graph for Active Directory",
+    "product_version_regex": ".*",
+    "publisher": "Splunk Community",
+    "contributors": [
```
Comment (dgopani-crest) [inline msadgraph.json:12]: This is certified app, please remove this

File: readme.html [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+      After making these changes, click on <b>Grant admin consent</b>.
+
+    <h2>Configure the MS Graph for Active Directory Phantom app Asset</h2>
+      When creating an asset for the <b>MS Graph for Active Directory</b> app, place the <b>Application ID</b> of the app created during the previous step in the <b>Client ID</b> field and place the password generated during the app creation process in the <b>Client Secret</b> field. Then, after filling out the <b>Tenant</b> field, click <b>SAVE</b>.
+      <br><br>
+      After saving, a new field will appear in the <b>Asset Settings</b> tab. Take the URL found in the <b>POST incoming for MS Graph to this location</b> field and place it in the <b>Redirect URIs</b> field mentioned in a previous step. To this URL, add <b>/result</b>. After doing so the URL should look something like:
+      <br><br>
+      <p>
+        https://&lt;phantom_host&gt;/rest/handler/msgraphforactivedirectory_c6d3b801-5c26-4abd-9e89-6d8007e2778f/&lt;asset_name&gt;/result
```
Comment (dgopani-crest) [inline readme.html:56]: msgraphforactivedirectory_f2a239df-acb2-47d6-861c-726a435cfe76

File: msadgraph.json [hunk 5]
```diff
@@ -0,0 +1,3154 @@
...
+                }
+            ],
+            "render": {
+                "type": "table"
+            },
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "list directory roles",
+            "description": "List the directory roles in a tenant",
```
Comment (dgopani-crest) [inline msadgraph.json:2880]: List the directory roles that are activated in the tenant

File: msadgraph.json [hunk 6]
```diff
@@ -0,0 +1,3154 @@
+{
+    "appid": "f2a239df-acb2-47d6-861c-726a435cfe76",
+    "name": "MS Graph for Active Directory",
+    "description": "Connects to Microsoft AD using MS Graph REST API services",
+    "type": "identity management",
+    "product_vendor": "Microsoft",
+    "logo": "logo_msadgraph_dark.svg",
```
Comment (dgopani-crest) [inline msadgraph.json:7]: logo_msadgraph.svg

File: msadgraph.json [hunk 7]
```diff
@@ -0,0 +1,3154 @@
+{
+    "appid": "f2a239df-acb2-47d6-861c-726a435cfe76",
+    "name": "MS Graph for Active Directory",
+    "description": "Connects to Microsoft AD using MS Graph REST API services",
```
Comment (dgopani-crest) [inline msadgraph.json:4]: Connects to Microsoft Active Directory using MS Graph REST API services

File: msadgraph.json (hunk 2 above)
Comment (ebhoir-crest) [reply msadgraph.json]: updated copyrights

File: release_notes/unreleased.md [hunk 8]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Initial release
```
Comment (dgopani-crest) [inline release_notes/unreleased.md:2]: Initial release with Python3 support

Also, add notes like  "This app is a replacement for the Azure AD app with Microsoft Graph API as Azure AD API is going to deprecate in near future."

**Comment by dgopani-crest** [comment]
@ebhoir-crest remove all the .DS_Store files

File: __init__.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# File: __init__.py
+#
+# Copyright (c) 2019-2022 Splunk Inc.
```
Comment (dgopani-crest) [inline __init__.py:3]: it should be 2022 in all the files

Comment (ebhoir-crest) [reply __init__.py]: updated to 2022

File: release_notes/unreleased.md (hunk 8 above)
Comment (ebhoir-crest) [reply release_notes/unreleased.md]: updated as suggested

**Comment by ricardo-phantom** [comment]
There seems to be references to the word "phantom" in almost every file. I think we should probably switch to "soar" instead but probably worth to open the discussion to rest of the team here.

**Comment by dgopani-crest** [comment]
> There seems to be references to the word "phantom" in almost every file. I think we should probably switch to "soar" instead but probably worth to open the discussion to rest of the team here.

@ricardo-phantom Agree and Sounds good to me.
@mattsayar-splunk @brian-phantom @pzhou-splunk Can you share your thought on this?

**Comment by brian-phantom** [comment]
> > There seems to be references to the word "phantom" in almost every file. I think we should probably switch to "soar" instead but probably worth to open the discussion to rest of the team here.
> 
> @ricardo-phantom Agree and Sounds good to me. @mattsayar-splunk @brian-phantom @pzhou-splunk Can you share your thought on this?

I think that's what we should be doing at this point replace Phantom with SOAR.

**Review by brian-phantom — APPROVED** [review]

**Pull request merged**